cnd/src/network/comit.rs

```diff
@@ -540,90 +512,111 @@ impl NetworkBehaviourEventProcess<oneshot_behaviour::OutEvent<finalize::Message>
 
         if state.sent_finalized && state.received_finalized {
             tracing::info!("Swap {} is finalized.", swap_id);
-            let (local_swap_id, create_swap_params) = self
+            let (local_swap_id, _local_data) = self
                 .swaps
                 .finalize_swap(&swap_id)
-                .expect("Swap should be known");
+                .expect("swap should be known");
 
-            let secret_hash = self
-                .secret_hashes
+            let remote_data = self
+                .remote_data
                 .get(&swap_id)
-                .copied()
-                .expect("must exist");
-
-            let ethereum_identity = self.ethereum_identities.get(&swap_id).copied().unwrap();
-            let lightning_identity = self.lightning_identities.get(&swap_id).copied().unwrap();
+                .cloned()
+                .expect("valid remote data");
 
-            self.events.push_back(BehaviourOutEvent::SwapFinalized {
+            self.events.push_back(BehaviourOutEvent {
                 local_swap_id,
-                swap_params: create_swap_params,
-                secret_hash,
-                ethereum_identity,
-                lightning_identity,
+                remote_data,
             });
         }
     }
 }
 
+/// All possible data to be sent to the remote node for any protocol
+/// combination.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct LocalData {
```

Comment:
:question: Do you expect this to eventually be parametrized by alpha identity and beta identity?

Reply:
No, that's the big win with this PR - the comit module no longer knows about alpa and beta.

Reply:
Hmm, that's interesting. Why do you think it's better for it to know about optional concrete types rather than protocols?

There seems to be consensus around this being a positive change, but I'm always scared of structs with lots of optional fields. Maybe I don't understand the pain points of the other approach.

Reply:
Its because now `Comit` just handles `Option<identity::Lightning>`. by 'handles' we mean 'if this is Some value then send it to the peer'. `Comit` never needs to know _why_ there is a lightning identity i.e., `Comit` does not need extending for any protocol that includes a lightning identity. If its there we send it, if not we don't. Period.

In other words, once we add `Option<identity::hbit>` to the `Identities` struct then `Comit` can handle any combination of the three protocols without specifically ever knowing which protocol it is handling, this simplifies the code a whole bunch.

Reply:
`Comit` here refers to the comit network behaviour

Reply:
It does help! It makes sense now :)